### PR TITLE
cmake: cleanup

### PR DESCRIPTION
**Pull request by zonr**
- Remove a redundant check on netdb.h (EVENT__HAVE_NETDB_H).
  - Properly setup EVENT__SIZEOF_SSIZE_T for the case where "ssize_t"
    is not "int".
  - Remove unused EVENT__HAVE_PTHREAD.
  - Set EVENT__HAVE_LIBZ instead of EVENT__HAVE_ZLIB (and remove the
    latter).

### CODE REVIEW

**Comment by azat** [comment]
@zonr thanks for the patch and sorry for such a long delay!
I'm going to rebase it a little bit and apply.

File: CMakeLists.txt [hunk 1]
```diff
@@ -438,8 +437,10 @@ CHECK_TYPE_SIZE("SSIZE_T" EVENT__SIZEOF_SSIZE_T_UPPER LANGUAGE C)
 
 if (EVENT__SIZEOF_SSIZE_T_LOWER)
     set(EVENT__ssize_t "ssize_t")
+    set(EVENT__SIZEOF_SSIZE_T ${EVENT__SIZEOF_SSIZE_T_LOWER})
```
Comment (azat) [inline CMakeLists.txt]: `EVENT__SIZEOF_SSIZE_T` didn't used, so I'm going to remove it.

Comment (azat) [inline CMakeLists.txt]: Nope, I'm going to leave it as-is, since we have `EVENT__SIZEOF_OFF_T`, sorry for the noise

**Comment by zonr** [comment]
Thanks for your hard working on reviewing the patch. Please go ahead with changes as appropriated and feel free to let me know if you need any help.

**Comment by azat** [comment]
@zonr applied, thanks!

**Pull request closed without merging**